scipy/optimize/_optimize.py

```diff
@@ -474,6 +477,7 @@ def rosen_hess(x):
     return H + xpx.create_diagonal(diagonal, xp=xp)
 
 
+@xp_capabilities(skip_backends=[('jax', "JAX doesn't allow item assignment.")])
```

Comment:
Since `rosen_hess` and `rosen_hess_prod` share tests, and if we expect that they will be subject to the same backend limitations, should the two functions share an entry in the capabilities table? For example,
```suggestion
@xp_capabilities('rosen_hess',
                 skip_backends=[('jax', "JAX doesn't allow item assignment.")])
```